cg/services/order_validation_service/order_validation_service.py

```diff
@@ -1,9 +1,7 @@
 from abc import ABC, abstractmethod
 
-from cg.services.order_validation_service.models.errors import ValidationErrors
-
 
 class OrderValidationService(ABC):
     @abstractmethod
-    def validate(self, order_json: str) -> ValidationErrors:
+    def validate(self, order_json: str) -> dict:
```

Comment:
also, you should start thinking about where in the order submission this is supposed to go. Will this really be a JSON or something of type OrderIn that is created in the submit_order endpoint

Reply:
Good point! We should think about that, but we have a lot of work to do still before starting to integrate it.

Maybe we should return something indicating whether the validation passed/failed along side the validation response. Something like
```python
class ValidationResult(BaseModel):
    passed: bool
    data: dict

Reply:
I do think that this type hint is outdated, it should be a dict in my mind. I would also like us to move away from the OrderIn object at some point so I do not think it is necessarily useful to constrain ourselves to it. In practice, the TomteOrder, TomteCase and TomteSample models in this project should be better suited and swapped in for the old OrderIn objects. 

We do need to put some thought into how much would need to change in the submit_order endpoint for that to happen. But I really do not wish the OrderIn objects to stay around for long (they do not even reflect the structure in the frontend and seem only built to be flexible for each workflow). Also, I do not think there should be any logic converting the request to Pydantic models in the endpoint, especially since that is part of the validation. So I see no world in which we would want a pydantic model as input to this method. But there are indeed areas which we need to focus on with regards to the existing flow.

Reply:
We would ideally like to rewrite the submit_order logic altogether.

Reply:
Indeed, not for this PR.

The validate function could also return nothing and raise an error that can be caught in the relevant place.

Reply:
That is also an option 👍 As long as we cleanly encapsulate any logic related to the validation process in the validation service.

Reply:
Also - we do not wish to raise exceptions in this case, actually. The response will be used in the frontend to showcase to any customers what they need to change. So we would need to return a mapped error message to the correct part of the request stating what is wrong with it.

Reply:
I will resolve for now since it is out of scope for this PR.